sympy/solvers/solvers.py

```diff
@@ -999,6 +999,9 @@ def _sympified_list(w):
                     fi = fi.rewrite(Add, evaluate=False)
             f[i] = fi
 
+        if fi.has(binomial):
+            f[i] = fi.replace(lambda exp: binomial, lambda exp: expand_func(exp))
```

Comment:
The filter function here will return true for any expression:
```
In [1]: f = lambda exp: binomial                                                                                                               

In [2]: f(sin(x))                                                                                                                              
Out[2]: binomial

In [3]: bool(f(sin(x)))                                                                                                                        
Out[3]: True
```

Reply:
I have made the change for it to correctly identify the binomials.